### PR TITLE
Specify window.open monkeypatch

### PR DESCRIPTION
**Pull request by csharrison**
Note: this leaves an open issue after parsing the `attributionsrc` string similar to the anchor registration.

***
<a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/pull/476.html" title="Last updated on Jun 6, 2022, 3:37 PM UTC (757d4b1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/476/6d3b06c...757d4b1.html" title="Last updated on Jun 6, 2022, 3:37 PM UTC (757d4b1)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -206,6 +206,27 @@ Modify the step:
 
 to call <a spec="html">navigate</a> with |attributionSource| set to |attributionSource|.
 
+<h3 id="monkeypatch-window-open">Window open steps</h4>
+
+Attribution source information declared via `window.open()` need to be passed to
```
Comment (apasel422) [inline index.bs:211]:
```suggestion
Attribution source information declared via `window.open()` needs to be passed to
```

File: index.bs [hunk 2]
```diff
@@ -573,6 +594,19 @@ To <dfn>obtain an attribution source from an anchor</dfn> given an <{a}> element
...
 Issue: Specify the steps for making attributionsrc request.
 
+<h3 algorithm id="obtaining-attribution-source-window-features">Obtaining an attribution source from window features</h3>
+
+To <dfn>obtain an attribution source from window features</dfn> given an [=ordered map=] |tokenizedFeatures|
+and a [=browsing context=] |sourceBrowsingContext|:
+
+1. If |tokenizedFeatures|["attributionsrc"] does not exist, return null.
+1. If |sourceBrowsingContext|'s [=browsing context/active window=] does not have [=transient activation=], return null.
+1. Let |decodedSrcBytes| be the result of [=string/percent-decode|percent-decoding=] |tokenizedFeatures|["attributionsrc"]
```
Comment (apasel422) [inline index.bs:604]:
```suggestion
1. Let |decodedSrcBytes| be the result of [=string/percent-decode|percent-decoding=] |tokenizedFeatures|["attributionsrc"].
```

File: index.bs [hunk 3]
```diff
@@ -573,6 +594,19 @@ To <dfn>obtain an attribution source from an anchor</dfn> given an <{a}> element
...
+1. Let |decodedSrc| be the [=UTF-8 decode without BOM=] of |decodedSrcBytes|
```
Comment (apasel422) [inline index.bs:605]:
```suggestion
1. Let |decodedSrc| be the [=UTF-8 decode without BOM=] of |decodedSrcBytes|.
```

File: index.bs [hunk 4]
```diff
@@ -206,6 +206,27 @@ Modify the step:
...
+
+Attribution source information declared via `window.open()` need to be passed to
+the <a spec="html">navigate</a> algorithm.
+
+In <a spec="html">window open steps</a> after
+
+> 10. If target browsing context is null, then return null.
+
+add the following steps:
+1. Let <var>attributionSource</var> be the result of running [=obtain an attribution source from window features=] with tokenizedFeatures and source browsing context
```
Comment (apasel422) [inline index.bs:219]:
```suggestion
1. Let |attributionSource| be the result of running [=obtain an attribution source from window features=] with |tokenizedFeatures| and |source browsing context|.
```

File: index.bs [hunk 5]
```diff
@@ -206,6 +206,27 @@ Modify the step:
...
+
+Modify the step:
+> 3. Navigate target browsing context to request, with exceptionsEnabled set to true and the source browsing context set to source browsing context.
```
Comment (apasel422) [inline index.bs:222]:
```suggestion
> 3. Navigate |target browsing context| to request, with exceptionsEnabled set to true and the source browsing context set to source browsing context.
```
```suggestion
> 3. Navigate target browsing context to request, with exceptionsEnabled set to true and the source browsing context set to |source browsing context|.
```

Comment (apasel422) [inline index.bs:222]: We should also add the cross-reference for `exceptionsEnabled` and `the source browsing context` as in the spec itself.

File: index.bs [hunk 6]
```diff
@@ -573,6 +594,19 @@ To <dfn>obtain an attribution source from an anchor</dfn> given an <{a}> element
 
 Issue: Specify the steps for making attributionsrc request.
 
+<h3 algorithm id="obtaining-attribution-source-window-features">Obtaining an attribution source from window features</h3>
+
+To <dfn>obtain an attribution source from window features</dfn> given an [=ordered map=] |tokenizedFeatures|
+and a [=browsing context=] |sourceBrowsingContext|:
+
+1. If |tokenizedFeatures|["attributionsrc"] does not exist, return null.
```
Comment (apasel422) [inline index.bs:602]:
```suggestion
1. If |tokenizedFeatures|["attributionsrc"] does not [=map/exists|exist=], return null.
```

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: index.bs (hunk 5 above)
Comment (csharrison) [reply index.bs]: Done

**Comment by csharrison** [comment]
PTAL

**Pull request merged**